src/tree/AzureDBAPIStep.ts

```diff
@@ -0,0 +1,67 @@
+import { PostgresServerCreateStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCreateStep';
+import { PostgresServerCredPWStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCredPWStep';
+import { PostgresServerCredUserStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerCredUserStep';
+import { PostgresServerFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerFirewallStep';
+import { PostgresServerNameStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerNameStep';
+import { PostgresServerSetCredentialsStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetCredentialsStep';
+import { PostgresServerSetFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetFirewallStep';
+import { CosmosDBAccountCreateStep } from './CosmosDBAccountWizard/CosmosDBAccountCreateStep';
+import { CosmosDBAccountNameStep } from './CosmosDBAccountWizard/CosmosDBAccountNameStep';
+import { ICosmosDBWizardContext } from './CosmosDBAccountWizard/ICosmosDBWizardContext';
+import { IAzureDBWizardContext } from './IAzureDBWizardContext';
+
+export class AzureDBAPIStep extends AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext> {
+    public async prompt(wizardContext: IAzureDBWizardContext): Promise<void> {
+        const picks: IAzureQuickPickItem<Experience>[] = getExperienceQuickPicks();
+
+        const result: IAzureQuickPickItem<Experience> = await ext.ui.showQuickPick(picks, {
+            placeHolder: "Select an Azure Database Resource."
+        });
+
+        wizardContext.defaultExperience = result.data;
+    }
+
+    public async getSubWizard(wizardContext: IAzureDBWizardContext): Promise<IWizardOptions<IPostgresWizardContext | ICosmosDBWizardContext>> {
+        if (wizardContext.defaultExperience?.api === API.Postgres) {
+            // tslint:disable-next-line: no-shadowed-variable
+            const promptSteps: AzureWizardPromptStep<IPostgresWizardContext>[] = [
+                new PostgresServerNameStep(),
+                new PostgresServerCredUserStep(),
+                new PostgresServerCredPWStep(),
+                new PostgresServerConfirmPWStep(),
+                new PostgresServerFirewallStep()
+            ];
+            // tslint:disable-next-line: no-shadowed-variable
+            const executeSteps: AzureWizardExecuteStep<IPostgresWizardContext>[] = [
+                new PostgresServerCreateStep(),
+                new PostgresServerSetCredentialsStep(),
+                new PostgresServerSetFirewallStep()
+            ];
+            // tslint:disable-next-line: no-unnecessary-local-variable
+            const wizardOptions: IWizardOptions<IPostgresWizardContext> = { promptSteps, executeSteps };
+            return wizardOptions;
+        }
+        const promptSteps: AzureWizardPromptStep<ICosmosDBWizardContext>[] = [
```

Comment:
I'm a big fan of else cases and would suggest you almost always use one in situations like this. For example, if you forgot the line `return wizardOptions;` above, today it would fall through and return the cosmos stuff instead. If you used an else case and forgot that line, you would get a build/lint error saying this method doesn't always return a value. The build error makes it much easier to prevent bugs before anyone even starts testing

It would also fix your tslint errors about shadowed variables. I don't think I would ever disable that rule - I would always refactor to fix it